optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/score/director/TestdataComparableDifferentValuesCalculator.java

```diff
@@ -0,0 +1,34 @@
+package org.optaplanner.core.impl.testdata.score.director;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+import org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator;
+import org.optaplanner.core.impl.testdata.domain.TestdataValue;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataEntityWithDifficultyComparator;
+import org.optaplanner.core.impl.testdata.domain.comparable.TestdataSolutionWithDifficultyComparatorEntity;
+
+public class TestdataComparableDifferentValuesCalculator implements EasyScoreCalculator<TestdataSolutionWithDifficultyComparatorEntity> {
+
+    @Override
+    public SimpleScore calculateScore(TestdataSolutionWithDifficultyComparatorEntity solution) {
+        int score = 0;
+        Map<TestdataValue, Integer> alreadyUsedValues = new HashMap<>();
```

Comment:
Why is this a Map when a Set suffices?
The value side is never read. There is never a "get(key)", only a "contains(key").

Reply:
This one was a leftover from when I was trying to make a more complex Calculator in order to make the solver do more different steps, but after trying out a simple Set-based one and not seeing any benefit in having the previous one, I've changed it to Set as you've recommended, thank you.